butane_core/src/db/mod.rs

```diff
@@ -136,12 +541,7 @@ fn conn_complete_if_dir(path: &Path) -> Cow<Path> {
 }
 
 /// Database backend. A boxed implementation can be returned by name via [`get_backend`].
-pub trait Backend {
-    fn name(&self) -> &'static str;
-    fn create_migration_sql(&self, current: &adb::ADB, ops: Vec<adb::Operation>) -> Result<String>;
-    fn connect(&self, conn_str: &str) -> Result<Connection>;
-}
-
+#[async_trait]
 impl Backend for Box<dyn Backend> {
```

Comment:
The comment here on line 543 is out of place, as the `pub trait` was moved higher in the module to line 489.

as much of the trait has not changed, IMO it should be moved back here, immediately above the `impl`